Support RTL by auto detect direction

ATofighi opened the pull request:
With this change, the browser detect the direction of each element of editor, so If we write in RTL language in a paragraph, the direction change automatically to RTL for that paragraph.

I test it only using inspect element and add this style.

The result is here, paragraphs with RTL text automatically aligned from right and the direction is correct for both LTR and RTL texts.

![image](https://github.com/user-attachments/assets/b89ea7fc-7370-4155-9c0c-8a772ddcbb48)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/docmost/docmost?pullRequest=206) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/docmost/docmost?pullRequest=206) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/docmost/docmost?pullRequest=206) it.</sub>

Philipinho commented:
I hesitate because I am unsure if  `unicode-bidi: plaintext` would break anything. 
I am also considering this Tiptap extension https://github.com/amirhhashemi/tiptap-text-direction.

ATofighi committed:
Support RTL: auto set direction of bidi text

 - TextDirection extendion in editor
 - text-align: start in styles
 - unicode-bidi: plaintext in page tree

ATofighi commented:
> I hesitate because I am unsure if `unicode-bidi: plaintext` would break anything. I am also considering this Tiptap extension https://github.com/amirhhashemi/tiptap-text-direction.

I have changed the code based on your advise and used tiptap-text-direction extension.

But in title editor and page titles, unfortunately, I can't adjust perfectly with this extension, in these two places, I use the unicode-bidi: plaintext for direction: auto.

Can you review the code again please?

ATofighi commented:
@Philipinho I change the code. Can you review it please?

Philipinho commented:
Hi @ATofighi, thank you for your work. I will find some time to test it soon.
I appreciate your patience.

Philipinho committed:
add text-direction extension to backend schema

Philipinho commented:
Are you able to use the forward slash "/"?

ATofighi commented:
> Are you able to use the forward slash "/"?

yes.
![image](https://github.com/user-attachments/assets/9c6aaa8e-8124-4409-8568-23802af7d9c9)


Is forward slash not working for you?

Philipinho commented:
Sorry. I mean when the keyboard is set a RTL language like Arabic. `"/"` gives me `"\"` and `"\"` gives me `"ء"` on Mac.

ATofighi commented:
> Sorry. I mean when the keyboard is set a RTL language like Arabic. `"/"` gives me `"\"` and `"\"` gives me `"ء"` on Mac.


My main language is Persian and it has "/" in the keyboard layout, I asked chatgpt and it says that Arabic's keyboard layout also has "/" character, but its position may be differ from English.

ATofighi commented:
> Sorry. I mean when the keyboard is set a RTL language like Arabic. `"/"` gives me `"\"` and `"\"` gives me `"ء"` on Mac.

My OS is Ubuntu, Shift + L is slash for me in Arabic keyboad

mikinushu commented:
Is this also relevant to the Hebrew language? Because it's still left to right.. 🤔
![image](https://github.com/user-attachments/assets/87cf55a4-9935-483b-b47f-8325c9da97e5)

ATofighi commented:
> Is this also relevant to the Hebrew language? Because it's still left to right.. 🤔 ![image](https://private-user-images.githubusercontent.com/6978721/413967040-87cf55a4-9935-483b-b47f-8325c9da97e5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzk4MTk2MTEsIm5iZiI6MTczOTgxOTMxMSwicGF0aCI6Ii82OTc4NzIxLzQxMzk2NzA0MC04N2NmNTVhNC05OTM1LTQ4M2ItYjQ3Zi04MzI1YzlkYTk3ZTUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDIxNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAyMTdUMTkwODMxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTFhYmQxNTNhNjA3NjVmMjY5OThjNTY5ZDk2M2JlOTNhYmJmNzlmOGRjZmFiZDFiODQ3OWYxMDcxOTJmZDU0OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.ToqqdiB0bZ9l-LAmMOB7egdhxkkcLS4g-P6IpQMQC_A)

I think it should be. can you give me an example text that help me to reproduce the issue?

mikinushu commented:
Here:
שורה עם כמה מילים בעברית

Philipinho commented:
Not yet merged due to a reported bug in the text-direction extension https://github.com/amirhhashemi/tiptap-text-direction/issues/18 which may affect Japanese users.

ATofighi commented:
> שורה עם כמה מילים בעברית

Are you sure that you test the code of this branch? I test "שורה עם כמה מילים בעברית" and it become RTL!

ATofighi commented:
> Not yet merged due to a reported bug in the text-direction extension [amirhhashemi/tiptap-text-direction#18](https://github.com/amirhhashemi/tiptap-text-direction/issues/18) which may affect Japanese users.

Thanks, I suggest a fix to the tiptap-text-direction author for this issue.

ATofighi committed:
temp: use a custom code for tiptap-text-direction to fix japanese ime conflict

ATofighi commented:
I also add an temporary commit that move the code of plugin to the project and do my fix on it to have a bug free version.

neuodev commented:
I hope that this pull request will be merged soon.

ATofighi commented:
> @Philipinho It seems that maintainer of https://github.com/amirhhashemi/tiptap-text-direction isn't available. May I copy the fixed version of plugin to docmost repository and create a fully functional PR without dependency on https://github.com/amirhhashemi/tiptap-text-direction ?

@Philipinho ? :grin:
